kotskinds/apis/kots/v1beta1/config_types.go

```diff
@@ -56,12 +56,26 @@ type ConfigItem struct {
 	CountByGroup  map[string]int         `json:"countByGroup,omitempty"`
 	Templates     []RepeatTemplate       `json:"templates,omitempty"`
 	ValuesByGroup ValuesByGroup          `json:"valuesByGroup,omitempty"`
+	Validation    *ConfigItemValidation  `json:"validation,omitempty"`
 	// Props       map[string]interface{} `json:"props,omitempty"`
 	// DefaultCmd  *ConfigItemCmd         `json:"default_cmd,omitempty"`
 	// ValueCmd    *ConfigItemCmd         `json:"value_cmd,omitempty"`
 	// DataCmd     *ConfigItemCmd         `json:"data_cmd,omitempty"`
 }
 
+type ConfigItemValidation struct {
+	Regex *RegexValidator `json:"regex,omitempty"`
+}
+
+type BaseValidator struct {
```

Comment:
Why is this base validator needed? It's not helping to implement an interface and it's not adding much value here. I'd suggest just moving the `Message` field to the `RegexValidator`

Reply:
When there are more validators, and if there are common fields, they can be added here.
Maybe this is to early to optimize. I will remove it.